### PR TITLE
Clean paradigm

### PR DESCRIPTION
**Pull request by bthirion**
Addresses issue https://github.com/nipy/nipy/issues/310

### CODE REVIEW

File: nipy/modalities/fmri/experimental_paradigm.py [hunk 1]
```diff
@@ -51,11 +52,16 @@ def __init__(self, con_id=None, onset=None, amplitude=None):
             try:
                 # this is only for backward compatibility:
                 #if con_id were integers, they become a string
+                warnings.warn(
+                    'The behavior will change,' +
+                    'prepending by a c will disappear in the future')
                 self.con_id = np.array(['c' + str(int(float(c)))
-                                        for c in con_id])
+                                         for c in con_id])
             except:
```
Comment (matthew-brett) [inline nipy/modalities/fmri/experimental_paradigm.py]: What error are you trying to catch with the `except`?   Should try and avoid bare `except` if at all possible.

File: nipy/modalities/fmri/tests/test_paradigm.py [hunk 2]
```diff
@@ -77,6 +77,15 @@ def test_read_paradigm():
     assert (read_paradigm.onset == paradigm.onset).all()
 
 
+def test_paradigm_with_int_condition_ids():
+    paradigm1 = basic_paradigm()
+    conditions = [0, 0, 0, 1, 1, 1, 2, 2, 2]
+    paradigm2 =  EventRelatedParadigm(conditions, paradigm1.onset)
+    #assert (paradigm2.con_id == np.array(conditions).astype('str')).all()
```
Comment (matthew-brett) [inline nipy/modalities/fmri/tests/test_paradigm.py]: Remove comment?  Or expand to say why comment remains?

Comment (bthirion) [inline nipy/modalities/fmri/tests/test_paradigm.py]: You're right on both points. I have edited the code accordingly (after a rebase).
Best,

File: nipy/modalities/fmri/experimental_paradigm.py [hunk 3]
```diff
@@ -48,13 +48,7 @@ def __init__(self, con_id=None, onset=None, amplitude=None):
         self.amplitude = amplitude
         if con_id is not None:
             self.n_events = len(con_id)
-            try:
```
Comment (matthew-brett) [inline nipy/modalities/fmri/experimental_paradigm.py]: I'm still worried by backwards compatibility - I think previous behavior was:

```
import nipy.modalities.fmri.experimental_paradigm as ep
In [2]: ep.Paradigm([1, 2, 3]).con_id
Out[2]: 
array(['c1', 'c2', 'c3'], 
      dtype='|S2')
```

where behavior on this branch is:

```
In [3]: ep.Paradigm([1, 2, 3]).con_id
Out[3]: 
array(['1', '2', '3'], 
      dtype='|S1')
```

I can see that it's a good idea not to convert strings to ints then back to strings, but maybe preserve `c` prefix if condition names are integers?

Comment (bthirion) [inline nipy/modalities/fmri/experimental_paradigm.py]: I think that in general it is a bad idea to decide that the conditions cannot be named '1', '2' etc. and to change this for the user. Now, if you think that backward compatibility is more important, I can revert that one.

Comment (matthew-brett) [inline nipy/modalities/fmri/experimental_paradigm.py]: It's OK, I can see arguments both ways...

**Comment by matthew-brett** [comment]
Thanks - sorry to be slow in merging.

**Pull request merged**